### PR TITLE
Fix last day of month handling logic in tests

### PR DESCRIPTION
**Pull request by hariombalhara**
Fix getSchedule tests, end of month is tricky.

### CODE REVIEW

**Comment by PeerRich** [comment]
@emrysal can you review this?

File: apps/web/test/lib/getSchedule.test.ts [hunk 1]
```diff
@@ -46,8 +46,10 @@ const getDate = (param: { dateIncrement?: number; monthIncrement?: number; yearI
   let _month = new Date().getMonth() + monthIncrement + 1;
 
   // If last day of the month(As _month is plus 1 already it is going to be the 0th day of next month which is the last day of current month)
-  if (_date === new Date(year, _month, 0).getDate()) {
-    _date = 1;
+  const lastDayOfMonth = new Date(year, _month, 0).getDate();
+  const numberOfDaysInNextMonth = +_date - +lastDayOfMonth;
```
Comment (zomars) [inline apps/web/test/lib/getSchedule.test.ts:50]:
```suggestion
  const numberOfDaysForNextMonth = +_date - +lastDayOfMonth;
```
Shouldn't this be numberOfDaysForNextMonth?

Comment (hariombalhara) [reply apps/web/test/lib/getSchedule.test.ts:50]: Yeah that is a better name.

File: apps/web/test/lib/getSchedule.test.ts [hunk 2]
```diff
@@ -46,8 +46,10 @@ const getDate = (param: { dateIncrement?: number; monthIncrement?: number; yearI
   let _month = new Date().getMonth() + monthIncrement + 1;
 
   // If last day of the month(As _month is plus 1 already it is going to be the 0th day of next month which is the last day of current month)
-  if (_date === new Date(year, _month, 0).getDate()) {
-    _date = 1;
+  const lastDayOfMonth = new Date(year, _month, 0).getDate();
+  const numberOfDaysForNextMonth = +_date - +lastDayOfMonth;
+  if (numberOfDaysInNextMonth > 0) {
+    _date = numberOfDaysInNextMonth;
```
Comment (zomars) [inline apps/web/test/lib/getSchedule.test.ts:52]:
```suggestion
    _date = numberOfDaysForNextMonth;
```

File: apps/web/test/lib/getSchedule.test.ts [hunk 3]
```diff
@@ -46,8 +46,10 @@ const getDate = (param: { dateIncrement?: number; monthIncrement?: number; yearI
   let _month = new Date().getMonth() + monthIncrement + 1;
 
   // If last day of the month(As _month is plus 1 already it is going to be the 0th day of next month which is the last day of current month)
-  if (_date === new Date(year, _month, 0).getDate()) {
-    _date = 1;
+  const lastDayOfMonth = new Date(year, _month, 0).getDate();
+  const numberOfDaysForNextMonth = +_date - +lastDayOfMonth;
+  if (numberOfDaysInNextMonth > 0) {
```
Comment (zomars) [inline apps/web/test/lib/getSchedule.test.ts:51]:
```suggestion
  if (numberOfDaysForNextMonth > 0) {
```

**Review by zomars — APPROVED** [review]

**Pull request merged**